src/components/Map.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { Component } from 'react';
+import GoogleMapReact from 'google-map-react';
+
+import withResource from './withResource';
+import { TResource } from '../types';
+
+type Props = { data: TResource };
+
+const boulderCoordinates = {
+  lat: 40.0156852,
+  lng: -105.2792069
+};
+
+class Map extends Component<any, any> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      directionsAreToggled: false
+    };
+  }
+
+  toggleDirections = (resource: any) => {
+    let userLocation = this.props.currentUser.location;
+    let map = this.state.map;
+    let directionsDisplay = this.state.directionsDisplay;
+    let directionsService = this.state.directionsService;
+
+    if (this.state.directionsAreToggled == false) {
+      this.setState({ directionsAreToggled: true });
+
+      directionsDisplay.setMap(map);
+
+      let calculateAndDisplayRoute = (
+        directionsService: any,
+        directionsDisplay: any
+      ) => {
+        directionsService.route(
+          {
+            origin: userLocation,
+            destination: {
+              lat: resource.lat,
+              lng: resource.lng
+            },
+            travelMode: 'TRANSIT'
+          },
+          function(response: any, status: any) {
+            if (status === 'OK') {
+              directionsDisplay.setDirections(response);
+            } else {
+              window.alert('Directions request failed due to ' + status);
+            }
+          }
+        );
+      };
+
+      calculateAndDisplayRoute(directionsService, directionsDisplay);
+    } else {
+      directionsDisplay.setMap(null);
+      this.setState({ directionsAreToggled: false });
+    }
+  };
+
+  createMarkers = (resources: any, map: any, maps: any) => {
+    resources.forEach((resource: any) => {
+      let marker = new maps.Marker({
+        map: map,
+        title: resource.name,
+        position: {
+          lat: resource.lat,
+          lng: resource.lng
+        }
+      });
+
+      let infoWindow = new maps.InfoWindow({
+        content:
+          '<div><b>' +
```

Comment:
Any idea if this can accept jsx? If so, maybe a separate method could return the jsx for an info window.

Reply:
From what I can tell, it only accepts strings - can we stringify JSX?

Reply:
As far as I know, not without some kind of external library, but I don't know if it would be worth it at that point. I'm going to suggest adding a TODO comment to revisit this in the future.